ocaml/idl/dm_api.ml

```diff
@@ -444,4 +430,35 @@ let check api emergency_calls =
       )
       system
   in
+  (* Sanity check 9: New parameters must have a default value: we partially check
+     this by checking in the list of parameters, after we have seen a parameter with
+     a default value, other parameters after it must have defaults as well. Unfortunately
+     checking parameters with newer releases does not work well as there are existing
+     parameters with new releases but no default. *)
+  let _ =
+    let new_param_has_default obj_name msg_name ps =
```

Comment:
I think the "new parameter" could be defined as the ones whose release is larger than the release of its message/call.

Reply:
Yes, I totally agree that should be the proper way. But there are two ways of defining params, one with `versioned_params` and another just `params`, the `params` is not required to have default at all, and adding this check would break those existing params without defaults but have a later release than the message itself